test/jdk/java/net/httpclient/http3/H3QuicVTTest.java

```diff
@@ -0,0 +1,226 @@
+            throw new AssertionError("Unexpected null sslContext");
+        }
+        // create an H3 only server
+        h3Server = HttpTestServer.create(HTTP_3_URI_ONLY, sslContext);
+        h3Server.addHandler((exchange) -> exchange.sendResponseHeaders(200, 0), "/hello");
+        h3Server.start();
+        System.out.println("Server started at " + h3Server.getAddress());
+        requestURI = "https://" + h3Server.serverAuthority() + "/hello";
+    }
+
+    @AfterAll
+    public static void afterClass() throws Exception {
+        if (h3Server != null) {
+            System.out.println("Stopping server " + h3Server.getAddress());
+            h3Server.stop();
+        }
+    }
+
+    /**
+     * Issues various HTTP3 requests and verifies the responses are received
+     */
+    @Test
+    public void testBasicRequests() throws Exception {
+        try (final HttpClient client = newClientBuilderForH3()
+                .proxy(NO_PROXY)
+                .version(HTTP_3)
+                .sslContext(sslContext).build()) {
+            final URI reqURI = new URI(requestURI);
+            final HttpRequest.Builder reqBuilder = HttpRequest.newBuilder(reqURI)
+                    .version(HTTP_3)
+                    .setOption(H3_DISCOVERY, HTTP_3_URI_ONLY);
+
+            // GET
+            final HttpRequest req1 = reqBuilder.copy().GET().build();
+            System.out.println("\nIssuing request: " + req1);
+            final HttpResponse<Void> resp1 = client.send(req1, BodyHandlers.discarding());
+            Assertions.assertEquals(200, resp1.statusCode(), "unexpected response code for GET request");
+            assertSelectorThread(client);
+
+            // POST
+            final HttpRequest req2 = reqBuilder.copy().POST(BodyPublishers.ofString("foo")).build();
+            System.out.println("\nIssuing request: " + req2);
+            final HttpResponse<Void> resp2 = client.send(req2, BodyHandlers.discarding());
+            Assertions.assertEquals(200, resp2.statusCode(), "unexpected response code for POST request");
+            assertSelectorThread(client);
+
+            // HEAD
+            final HttpRequest req3 = reqBuilder.copy().HEAD().build();
+            System.out.println("\nIssuing request: " + req3);
+            final HttpResponse<Void> resp3 = client.send(req3, BodyHandlers.discarding());
+            Assertions.assertEquals(200, resp3.statusCode(), "unexpected response code for HEAD request");
+            assertSelectorThread(client);
+        }
+    }
+
+    void assertSelectorThread(HttpClient client) {
+        Objects.requireNonNull(client);
+        String clientId = client.toString().substring(client.toString().indexOf('('));
+        String name = "Thread(QuicSelector(HttpClientImpl" + clientId + "))";
+        Set<String> threads = new HashSet<>(Thread.getAllStackTraces().keySet().stream()
+                .map(Thread::getName)
+                .toList());
+        boolean found = threads.contains(name);
```

Comment:
Using thread name for this check looks very fragile. Can we devise a more programmatic method? E.g., ensuring no `StackTraceElement` contains `QuicSelectorThread`?